File: DataSummitSolution/DataSummitUI/src/app/efModels/blockPosition.ts
import { ProfileAttribute } from '../profileAttributes/models/profileAttribute';

export class BlockPosition {

	BlockPositionId: number;
	Name: string;
	CreatedDate?: Date | string;
	UserId?: number;
	ProfileAttributes: ProfileAttribute[];

	constructor(blockPositionId?: number, name?: string, createdDate?: Date | string,
		userId?: number, profileAttributes?: ProfileAttribute[])
	{
		this.BlockPositionId = blockPositionId || 0;
		this.Name = name || "";
		this.CreatedDate = createdDate || new Date(Date.now()) || null;
		this.UserId = userId || 0 || null;
		this.ProfileAttributes = profileAttributes;
	}
}

File: DataSummitSolution/DataSummitModels/DB/BlockPosition.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace DataSummitModels.DB
{
    public partial class BlockPosition
    {
        public BlockPosition()
        {
            TemplateAttributes = new HashSet<TemplateAttribute>();
        }

        public byte BlockPositionId { get; set; }
        public string Name { get; set; }
        public DateTime? CreatedDate { get; set; }
        public long? UserId { get; set; }

        public virtual ICollection<TemplateAttribute> TemplateAttributes { get; set; }
    }
}
